package.json

```diff
@@ -64,12 +64,14 @@
     "mocha": "^5.2.0",
     "node-dir": "^0.1.17",
     "npm-run-all": "^4.1.3",
+    "spawn-sync": "^2.0.0",
```

Comment:
there might be a missing peer-dep  in `os-locale`, without this I get:

```
WARNING in ./node_modules/os-locale/node_modules/cross-spawn/index.js 32:26-47
Module not found: Error: Can't resolve 'spawn-sync' in '/home/coder/git/conjure-typescript/node_modules/os-locale/node_modules/cross-spawn'
 @ ./node_modules/os-locale/node_modules/execa/index.js 4:19-41
 @ ./node_modules/os-locale/index.js 2:14-30
 @ ./node_modules/yargs/yargs.js 1154:23-43
 @ ./node_modules/yargs/index.js 5:14-32
 @ ./src/conjureTypeScript.ts 55:12-28
```

Reply:
It looks like `os-locale@2.1.0` -> `execa@0.7.0` ->  `cross-spawn@5.1.0`. [Here](https://github.com/moxystudio/node-cross-spawn/blob/1da4c09ccf658079849a3d191b16e59bc600e8b4/index.js#L26-L40) is the relevant bit of code. It looks like they're falling back to `spawn-sync` if `require("child_process").spawnSync` isn't defined. That was added in node 0.11.12 (see [here](https://nodejs.org/api/child_process.html#child_processspawnsynccommand-args-options) and [here](https://github.com/nodejs/node-v0.x-archive/blob/v0.11.12/ChangeLog#L7)). That was released in 2014... we should probably upgrade node instead 😬

Reply:
I am running node 12.x.x, not 100% sure but I think this is a problem regardless - since webpack will statically try to bundle `spawn-sync` regardless of the condition?

Reply:
Ah, that makes sense I didn't realize this was failing at build time. 

The other way you could handle this is to use the webpack ignore plugin: 

```js
plugins: [
    // ...

    // `spawn-sync` is an undeclared, optional dependency of `cross-spawn@5.1.0`
    new webpack.IgnorePlugin({ resourceRegExp: /^spawn-sync$/ }),        

    // ...
]
```

But it's probably safer to just take the dependency the way you have it in case some other package legitimately needs it (now or in the future).